services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/scanner/tables/TableFileHandleAssociationMapper.java

```diff
@@ -0,0 +1,79 @@
+package org.sagebionetworks.repo.manager.file.scanner.tables;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Set;
+
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.sagebionetworks.repo.manager.file.scanner.ScannedFileHandleAssociation;
+import org.sagebionetworks.repo.manager.table.TableEntityManager;
+import org.sagebionetworks.repo.model.dbo.dao.table.TableRowChangeUtils;
+import org.sagebionetworks.repo.model.dbo.persistence.table.DBOTableRowChange;
+import org.sagebionetworks.repo.model.jdo.KeyFactory;
+import org.sagebionetworks.repo.model.table.TableRowChange;
+import org.sagebionetworks.repo.web.NotFoundException;
+import org.sagebionetworks.table.model.SparseChangeSet;
+import org.sagebionetworks.workers.util.aws.message.RecoverableMessageException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.AmazonServiceException.ErrorType;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+
+@Service
+public class TableFileHandleAssociationMapper implements RowMapper<ScannedFileHandleAssociation> {
+	
+	private static final Logger LOG = LogManager.getLogger(TableFileHandleAssociationMapper.class);
+
+	private TableEntityManager tableManager;
+	
+	private static final RowMapper<DBOTableRowChange> DBO_MAPPER = new DBOTableRowChange().getTableMapping();
+	
+	@Autowired
+	public TableFileHandleAssociationMapper(TableEntityManager tableManager) {
+		this.tableManager = tableManager;
+	}
+
+	@Override
+	public ScannedFileHandleAssociation mapRow(ResultSet rs, int rowNum) throws SQLException {
+	
+		TableRowChange changeMetadata = TableRowChangeUtils.ceateDTOFromDBO(DBO_MAPPER.mapRow(rs, rowNum));
```

Comment:
constant number of table row changes should be fine.  Then for each row dynamically load sparse change.